<!doctype html><html lang=en-us><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=generator content="Source Themes Academic 4.7.0"><meta name=author content="Janani Ravi"><meta name=description content="Studying proteins through the lens of evolution can help identify conserved features and lineage-specific variants, and potentially, their functions. MolEvolvR (http://jravilab.org/molevolvr) is a web-app that enables researchers to run a general-purpose computational workflow for characterizing the molecular evolution and phylogeny of their proteins of interest. The web-app accepts input in multiple formats: protein/domain sequences, homologous proteins, or domain scans. MolEvolvR returns detailed homolog data, along with dynamic graphical summaries, e.g., MSA, phylogenetic trees, domain architectures, domain proximity networks, phyletic spreads, and co-occurrence patterns across lineages. Thus, MolEvolvR provides a powerful, easy-to-use interface for computational protein characterization."><link rel=alternate hreflang=en-us href=/publication/molevolvr_webapp/><meta name=theme-color content="hsl(339, 90%, 68%)"><script src=/js/mathjax-config.js></script>
<link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.0-1/css/all.min.css integrity="sha256-4w9DunooKSr3MFXHXWyFER38WmPdm361bQS/2KUWZbU=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/github.min.css crossorigin=anonymous title=hl-light disabled><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/dracula.min.css crossorigin=anonymous title=hl-dark><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css integrity="sha256-SHMGCYmST46SoyGgo4YR/9AlK1vf3ff84Aq9yK4hdqM=" crossorigin=anonymous><script src=https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin=anonymous async></script>
<script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js integrity crossorigin=anonymous async></script>
<link rel=stylesheet href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap"><link rel=stylesheet href=/css/academic.css><link rel=manifest href=/index.webmanifest><link rel=icon type=image/png href=/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_32x32_fill_lanczos_center_3.png><link rel=apple-touch-icon type=image/png href=/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_192x192_fill_lanczos_center_3.png><link rel=canonical href=/publication/molevolvr_webapp/><meta property="twitter:card" content="summary_large_image"><meta property="twitter:site" content="@JRaviLab"><meta property="twitter:creator" content="@JRaviLab"><meta property="og:site_name" content="JRaviLab"><meta property="og:url" content="/publication/molevolvr_webapp/"><meta property="og:title" content="MolEvolvR: a web-app for characterizing proteins using molecular evolution and phylogeny | JRaviLab"><meta property="og:description" content="Studying proteins through the lens of evolution can help identify conserved features and lineage-specific variants, and potentially, their functions. MolEvolvR (http://jravilab.org/molevolvr) is a web-app that enables researchers to run a general-purpose computational workflow for characterizing the molecular evolution and phylogeny of their proteins of interest. The web-app accepts input in multiple formats: protein/domain sequences, homologous proteins, or domain scans. MolEvolvR returns detailed homolog data, along with dynamic graphical summaries, e.g., MSA, phylogenetic trees, domain architectures, domain proximity networks, phyletic spreads, and co-occurrence patterns across lineages. Thus, MolEvolvR provides a powerful, easy-to-use interface for computational protein characterization."><meta property="og:image" content="/publication/molevolvr_webapp/featured.png"><meta property="twitter:image" content="/publication/molevolvr_webapp/featured.png"><meta property="og:locale" content="en-us"><meta property="article:published_time" content="2022-02-22T00:00:00+00:00"><meta property="article:modified_time" content="2022-02-22T00:00:00+00:00"><script type=application/ld+json>{"@context":"https://schema.org","@type":"Article","mainEntityOfPage":{"@type":"WebPage","@id":"/publication/molevolvr_webapp/"},"headline":"MolEvolvR: a web-app for characterizing proteins using molecular evolution and phylogeny","image":["/publication/molevolvr_webapp/featured.png"],"datePublished":"2022-02-22T00:00:00Z","dateModified":"2022-02-22T00:00:00Z","author":{"@type":"Person","name":"Jacob Krol"},"publisher":{"@type":"Organization","name":"JRaviLab","logo":{"@type":"ImageObject","url":"/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_192x192_fill_lanczos_center_3.png"}},"description":"Studying proteins through the lens of evolution can help identify conserved features and lineage-specific variants, and potentially, their functions. MolEvolvR (http://jravilab.org/molevolvr) is a web-app that enables researchers to run a general-purpose computational workflow for characterizing the molecular evolution and phylogeny of their proteins of interest. The web-app accepts input in multiple formats: protein/domain sequences, homologous proteins, or domain scans. MolEvolvR returns detailed homolog data, along with dynamic graphical summaries, e.g., MSA, phylogenetic trees, domain architectures, domain proximity networks, phyletic spreads, and co-occurrence patterns across lineages. Thus, MolEvolvR provides a powerful, easy-to-use interface for computational protein characterization."}</script><title>MolEvolvR: a web-app for characterizing proteins using molecular evolution and phylogeny | JRaviLab</title></head><body id=top data-spy=scroll data-offset=70 data-target=#TableOfContents class=dark><aside class=search-results id=search><div class=container><section class=search-header><div class="row no-gutters justify-content-between mb-3"><div class=col-6><h1>Search</h1></div><div class="col-6 col-search-close"><a class=js-search href=#><i class="fas fa-times-circle text-muted" aria-hidden=true></i></a></div></div><div id=search-box><input name=q id=search-query placeholder=Search... autocapitalize=off autocomplete=off autocorrect=off spellcheck=false type=search></div></section><section class=section-search-results><div id=search-hits></div></section></div></aside><nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main><div class=container><div class="d-none d-lg-inline-flex"><a class=navbar-brand href=/>JRaviLab</a></div><button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar aria-expanded=false aria-label="Toggle navigation">
<span><i class="fas fa-bars"></i></span></button><div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none"><a class=navbar-brand href=/>JRaviLab</a></div><div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content><ul class="navbar-nav d-md-inline-flex"><li class=nav-item><a class=nav-link href=/#about_group><span>Home</span></a></li><li class=nav-item><a class=nav-link href=/#interests><span>Interests</span></a></li><li class=nav-item><a class=nav-link href=/#news><span>News</span></a></li><li class=nav-item><a class=nav-link href=/#about><span>Team</span></a></li><li class=nav-item><a class=nav-link href=/#publications><span>Papers</span></a></li><li class=nav-item><a class=nav-link href=/#projects><span>Projects</span></a></li><li class=nav-item><a class=nav-link href=/#talks><span>Talks</span></a></li><li class=nav-item><a class=nav-link href=/#posts><span>Posts</span></a></li><li class=nav-item><a class=nav-link href=/#gallery><span>Photos</span></a></li><li class=nav-item><a class=nav-link href=/#contact><span>Contact</span></a></li></ul></div><ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2"><li class=nav-item><a class="nav-link js-search" href=#><i class="fas fa-search" aria-hidden=true></i></a></li><li class=nav-item><a class="nav-link js-dark-toggle" href=#><i class="fas fa-moon" aria-hidden=true></i></a></li></ul></div></nav><div class=pub><div class="article-container pt-3"><h1>MolEvolvR: a web-app for characterizing proteins using molecular evolution and phylogeny</h1><div class=article-metadata><div><span><a href=/authors/jacob-d-krol/>Jacob Krol</a></span>, <span><a href=/authors/joseph-t-burke/>Joseph Burke</a></span>, <span><a href=/authors/samuel-z-chen/>Samuel Chen</a></span>, <span><a href=/authors/lo-sosinski/>Lo Sosinski</a></span>, <span><a href=/authors/john-b-johnston/>John B Johnston</a></span>, <span><a href=/authors/admin/>Janani Ravi</a></span></div><span class=article-date>February 2022</span>
<span class=middot-divider></span>
<span class=article-categories><i class="fas fa-folder mr-1"></i><a href=/categories/research/>Research</a>, <a href=/categories/r/>R</a></span></div><div class="btn-links mb-3"><a class="btn btn-outline-primary my-1 mr-1" href=https://doi.org/10.1101/2022.02.18.461833 target=_blank rel=noopener>Preprint</a>
<a class="btn btn-outline-primary my-1 mr-1" href=https://www.biorxiv.org/content/10.1101/2022.02.18.461833v2.full.pdf target=_blank rel=noopener>PDF</a>
<a class="btn btn-outline-primary my-1 mr-1" href=/project/pathogenic-features/>Project</a>
<a class="btn btn-outline-primary my-1 mr-1" href=https://doi.org/10.1101/2022.02.18.461833 target=_blank rel=noopener>DOI</a>
<a class="btn btn-outline-primary my-1 mr-1" href=http://jravilab.org/molevolvr target=_blank rel=noopener>Webapp</a></div></div><div class="article-header article-container featured-image-wrapper mt-4 mb-4" style=max-width:720px;max-height:635px><div style=position:relative><img src=/publication/molevolvr_webapp/featured_hu81654f8948b60c27bf336a3353477e56_1684447_720x0_resize_lanczos_3.png alt class=featured-image></div></div><div class=article-container><h3>Abstract</h3><p class=pub-abstract>Studying proteins through the lens of evolution can help identify conserved features and lineage-specific variants, and potentially, their functions. MolEvolvR (
<a href=http://jravilab.org/molevolvr target=_blank rel=noopener>http://jravilab.org/molevolvr</a>) is a web-app that enables researchers to run a general-purpose computational workflow for characterizing the molecular evolution and phylogeny of their proteins of interest. The web-app accepts input in multiple formats: protein/domain sequences, homologous proteins, or domain scans. MolEvolvR returns detailed homolog data, along with dynamic graphical summaries, e.g., MSA, phylogenetic trees, domain architectures, domain proximity networks, phyletic spreads, and co-occurrence patterns across lineages. Thus, MolEvolvR provides a powerful, easy-to-use interface for computational protein characterization.</p><div class=row><div class=col-md-1></div><div class=col-md-10><div class=row><div class="col-12 col-md-3 pub-row-heading">Type</div><div class="col-12 col-md-9"><a href=/publication/#3>Preprint</a></div></div></div><div class=col-md-1></div></div><div class="d-md-none space-below"></div><div class=row><div class=col-md-1></div><div class=col-md-10><div class=row><div class="col-12 col-md-3 pub-row-heading">Publication</div><div class="col-12 col-md-9"><em>bioRxiv</em></div></div></div><div class=col-md-1></div></div><div class="d-md-none space-below"></div><div class=space-below></div><div class=article-style><p>JTB, SZC, LMS are co-primary authors who contributed equally; they are listed alphabetically.</p></div><div class=article-tags><a class="badge badge-light" href=/tags/pathogenic-features/>pathogenic-features</a>
<a class="badge badge-light" href=/tags/host/pathogen/>host/pathogen</a>
<a class="badge badge-light" href=/tags/molecular-evolution/>molecular-evolution</a>
<a class="badge badge-light" href=/tags/phylogeny/>phylogeny</a>
<a class="badge badge-light" href=/tags/tree-of-life/>tree-of-life</a>
<a class="badge badge-light" href=/tags/webapp/>webapp</a>
<a class="badge badge-light" href=/tags/compbio-bioinfo/>compbio-bioinfo</a>
<a class="badge badge-light" href=/tags/r/>R</a></div><div class="media author-card content-widget-hr"><img class="avatar mr-3 avatar-circle" src=/authors/jacob-d-krol/avatar_hu3d5c6bfe37a17c163ce59dd54d0d69cc_2454260_270x270_fill_lanczos_center_3.png alt=Avatar><div class=media-body><h5 class=card-title><a href=/authors/jacob-d-krol/>Jacob Krol</a></h5><h6 class=card-subtitle>Information Science Professional</h6><ul class=network-icon aria-hidden=true><li><a href=https://github.com/jakekrol target=_blank rel=noopener><i class="fab fa-github"></i></a></li><li><a href=https://www.linkedin.com/in/jacob-krol-b3b784156/ target=_blank rel=noopener><i class="fab fa-linkedin"></i></a></li></ul></div></div><div class="article-widget content-widget-hr"><h3>Related</h3><ul><li><a href=/publication/psp_evolution/>The phage shock protein (PSP) envelope stress response: discovery of novel partners and evolutionary history</a></li><li><a href=/publication/dcia_evolution/>DciA Helicase Operators Exhibit Diversity across Bacterial Phyla</a></li><li><a href=/publication/inlp_listeria/>Novel Internalin P homologs in Listeria ivanovii londoniensis and Listeria seeligeri</a></li><li><a href=/publication/avcd_vcholerae/>Phage defence by deaminase-mediated depletion of deoxynucleotides in bacteria</a></li><li><a href=/publication/slps_grampos/>S-layers: the proteinaceous multifunctional armours of Gram-positive pathogens</a></li></ul></div></div></div><script src=https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js integrity="sha256-eOgo0OtLL4cdq7RdwRUiGKLX9XsIJ7nGhWEKbohmVAQ=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/r.min.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.js integrity="sha256-EErZamuLefUnbMBQbsEqu1USa+btR2oIlCpBJbyD4/g=" crossorigin=anonymous></script>
<script>const code_highlighting=!0</script><script>const isSiteThemeDark=!0</script><script>const search_config={indexURI:"/index.json",minLength:1,threshold:.3},i18n={no_results:"No results found",placeholder:"Search...",results:"results found"},content_type={post:"Posts",project:"Projects",publication:"Publications",talk:"Talks"}</script><script id=search-hit-fuse-template type=text/x-template>
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script><script src=https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin=anonymous></script>
<script src=/js/academic.min.df22e735169cca39ea07a999d8cba15f.js></script><div class=container><footer class=site-footer><p class=powered-by>Â© Janani Ravi, 2025 &#183;
Powered by the
<a href=https://sourcethemes.com/academic/ target=_blank rel=noopener>Academic theme</a> for
<a href=https://gohugo.io target=_blank rel=noopener>Hugo</a>.
<span class=float-right aria-hidden=true><a href=# class=back-to-top><span class=button_icon><i class="fas fa-chevron-up fa-2x"></i></span></a></span></p></footer></div><div id=modal class="modal fade" role=dialog><div class=modal-dialog><div class=modal-content><div class=modal-header><h5 class=modal-title>Cite</h5><button type=button class=close data-dismiss=modal aria-label=Close>
<span aria-hidden=true>&#215;</span></button></div><div class=modal-body><pre><code class="tex hljs"></code></pre></div><div class=modal-footer><a class="btn btn-outline-primary my-1 js-copy-cite" href=# target=_blank><i class="fas fa-copy"></i> Copy</a>
<a class="btn btn-outline-primary my-1 js-download-cite" href=# target=_blank><i class="fas fa-download"></i> Download</a><div id=modal-error></div></div></div></div></div></body></html>