<!doctype html><html lang=en-us><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="IE=edge"><meta name=generator content="Source Themes Academic 4.7.0"><meta name=author content="Janani Ravi"><meta name=description content="Since the clinical introduction of antibiotics in the 1940s, antimicrobial resistance (AMR) has become an increasingly dire threat to global public health. Pathogens acquire AMR much faster than we discover new drugs, warranting new methods to better understand the molecular underpinnings of AMR. Traditional approaches for detecting AMR in novel bacterial strains are time-consuming and labor-intensive. However, advances in sequencing technology offer a plethora of bacterial genome data, and computational approaches like machine learning (ML) provide an optimistic scope for in silico AMR prediction. Here, we introduce a comprehensive multiscale ML approach to predict AMR phenotypes and identify AMR molecular features associated with a single drug or drug family, stratified by time and geographical locations. As a case study, we focus on a subset of the World Health Organization’s Bacterial Priority Pathogens, frequently drug-resistant and nosocomial bacteria: Enterococcus faecium, Staphylococcus aureus, Acinetobacter baumannii, Pseudomonas aeruginosa, and two Enterobacter species. We started with sequenced genomes with lab-derived AMR phenotypes, constructed pangenomes, clustered protein sequences, and extracted protein domains to generate pangenomic features across molecular scales. To uncover the molecular mechanisms behind drug-/drug family-specific resistance, we trained logistic regression ML models on our datasets. These yielded ranked lists of AMR-associated genes, proteins, and domains. In addition to recapitulating known AMR features, our models identified novel candidates for experimental validation. They performed well across molecular scales, data types (binary vs. counts), and drugs. The models achieved a median normalized Matthews correlation coefficient of 0.89, suggesting robust performance across molecular scales with counts or presence/absence of features. Model performance showed resilience even when evaluated on geographical and temporal holdouts. Our holistic approach promises reliable prediction of existing and developing resistance in newly sequenced pathogen genomes, along with the mechanistic molecular contributors of AMR."><link rel=alternate hreflang=en-us href=/publication/amr_eskape/><meta name=theme-color content="hsl(339, 90%, 68%)"><script src=/js/mathjax-config.js></script>
<link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.0-1/css/all.min.css integrity="sha256-4w9DunooKSr3MFXHXWyFER38WmPdm361bQS/2KUWZbU=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin=anonymous><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/github.min.css crossorigin=anonymous title=hl-light disabled><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/dracula.min.css crossorigin=anonymous title=hl-dark><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css integrity="sha256-SHMGCYmST46SoyGgo4YR/9AlK1vf3ff84Aq9yK4hdqM=" crossorigin=anonymous><script src=https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin=anonymous async></script>
<script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js integrity crossorigin=anonymous async></script>
<link rel=stylesheet href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap"><link rel=stylesheet href=/css/academic.css><link rel=manifest href=/index.webmanifest><link rel=icon type=image/png href=/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_32x32_fill_lanczos_center_3.png><link rel=apple-touch-icon type=image/png href=/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_192x192_fill_lanczos_center_3.png><link rel=canonical href=/publication/amr_eskape/><meta property="twitter:card" content="summary_large_image"><meta property="twitter:site" content="@JRaviLab"><meta property="twitter:creator" content="@JRaviLab"><meta property="og:site_name" content="JRaviLab"><meta property="og:url" content="/publication/amr_eskape/"><meta property="og:title" content="From sequence to signature: Uncovering multiscale AMR features across bacterial pathogens with supervised machine learning | JRaviLab"><meta property="og:description" content="Since the clinical introduction of antibiotics in the 1940s, antimicrobial resistance (AMR) has become an increasingly dire threat to global public health. Pathogens acquire AMR much faster than we discover new drugs, warranting new methods to better understand the molecular underpinnings of AMR. Traditional approaches for detecting AMR in novel bacterial strains are time-consuming and labor-intensive. However, advances in sequencing technology offer a plethora of bacterial genome data, and computational approaches like machine learning (ML) provide an optimistic scope for in silico AMR prediction. Here, we introduce a comprehensive multiscale ML approach to predict AMR phenotypes and identify AMR molecular features associated with a single drug or drug family, stratified by time and geographical locations. As a case study, we focus on a subset of the World Health Organization’s Bacterial Priority Pathogens, frequently drug-resistant and nosocomial bacteria: Enterococcus faecium, Staphylococcus aureus, Acinetobacter baumannii, Pseudomonas aeruginosa, and two Enterobacter species. We started with sequenced genomes with lab-derived AMR phenotypes, constructed pangenomes, clustered protein sequences, and extracted protein domains to generate pangenomic features across molecular scales. To uncover the molecular mechanisms behind drug-/drug family-specific resistance, we trained logistic regression ML models on our datasets. These yielded ranked lists of AMR-associated genes, proteins, and domains. In addition to recapitulating known AMR features, our models identified novel candidates for experimental validation. They performed well across molecular scales, data types (binary vs. counts), and drugs. The models achieved a median normalized Matthews correlation coefficient of 0.89, suggesting robust performance across molecular scales with counts or presence/absence of features. Model performance showed resilience even when evaluated on geographical and temporal holdouts. Our holistic approach promises reliable prediction of existing and developing resistance in newly sequenced pathogen genomes, along with the mechanistic molecular contributors of AMR."><meta property="og:image" content="/publication/amr_eskape/featured.png"><meta property="twitter:image" content="/publication/amr_eskape/featured.png"><meta property="og:locale" content="en-us"><meta property="article:published_time" content="2025-07-07T00:00:00+00:00"><meta property="article:modified_time" content="2025-07-07T00:00:00+00:00"><script type=application/ld+json>{"@context":"https://schema.org","@type":"Article","mainEntityOfPage":{"@type":"WebPage","@id":"/publication/amr_eskape/"},"headline":"From sequence to signature: Uncovering multiscale AMR features across bacterial pathogens with supervised machine learning","image":["/publication/amr_eskape/featured.png"],"datePublished":"2025-07-07T00:00:00Z","dateModified":"2025-07-07T00:00:00Z","author":{"@type":"Person","name":"Abhirupa Ghosh"},"publisher":{"@type":"Organization","name":"JRaviLab","logo":{"@type":"ImageObject","url":"/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_192x192_fill_lanczos_center_3.png"}},"description":"Since the clinical introduction of antibiotics in the 1940s, antimicrobial resistance (AMR) has become an increasingly dire threat to global public health. Pathogens acquire AMR much faster than we discover new drugs, warranting new methods to better understand the molecular underpinnings of AMR. Traditional approaches for detecting AMR in novel bacterial strains are time-consuming and labor-intensive. However, advances in sequencing technology offer a plethora of bacterial genome data, and computational approaches like machine learning (ML) provide an optimistic scope for in silico AMR prediction. Here, we introduce a comprehensive multiscale ML approach to predict AMR phenotypes and identify AMR molecular features associated with a single drug or drug family, stratified by time and geographical locations. As a case study, we focus on a subset of the World Health Organization’s Bacterial Priority Pathogens, frequently drug-resistant and nosocomial bacteria: Enterococcus faecium, Staphylococcus aureus, Acinetobacter baumannii, Pseudomonas aeruginosa, and two Enterobacter species. We started with sequenced genomes with lab-derived AMR phenotypes, constructed pangenomes, clustered protein sequences, and extracted protein domains to generate pangenomic features across molecular scales. To uncover the molecular mechanisms behind drug-/drug family-specific resistance, we trained logistic regression ML models on our datasets. These yielded ranked lists of AMR-associated genes, proteins, and domains. In addition to recapitulating known AMR features, our models identified novel candidates for experimental validation. They performed well across molecular scales, data types (binary vs. counts), and drugs. The models achieved a median normalized Matthews correlation coefficient of 0.89, suggesting robust performance across molecular scales with counts or presence/absence of features. Model performance showed resilience even when evaluated on geographical and temporal holdouts. Our holistic approach promises reliable prediction of existing and developing resistance in newly sequenced pathogen genomes, along with the mechanistic molecular contributors of AMR."}</script><title>From sequence to signature: Uncovering multiscale AMR features across bacterial pathogens with supervised machine learning | JRaviLab</title></head><body id=top data-spy=scroll data-offset=70 data-target=#TableOfContents class=dark><aside class=search-results id=search><div class=container><section class=search-header><div class="row no-gutters justify-content-between mb-3"><div class=col-6><h1>Search</h1></div><div class="col-6 col-search-close"><a class=js-search href=#><i class="fas fa-times-circle text-muted" aria-hidden=true></i></a></div></div><div id=search-box><input name=q id=search-query placeholder=Search... autocapitalize=off autocomplete=off autocorrect=off spellcheck=false type=search></div></section><section class=section-search-results><div id=search-hits></div></section></div></aside><nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id=navbar-main><div class=container><div class="d-none d-lg-inline-flex"><a class=navbar-brand href=/>JRaviLab</a></div><button type=button class=navbar-toggler data-toggle=collapse data-target=#navbar-content aria-controls=navbar aria-expanded=false aria-label="Toggle navigation">
<span><i class="fas fa-bars"></i></span></button><div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none"><a class=navbar-brand href=/>JRaviLab</a></div><div class="navbar-collapse main-menu-item collapse justify-content-start" id=navbar-content><ul class="navbar-nav d-md-inline-flex"><li class=nav-item><a class=nav-link href=/#about_group><span>Home</span></a></li><li class=nav-item><a class=nav-link href=/#interests><span>Interests</span></a></li><li class=nav-item><a class=nav-link href=/#news><span>News</span></a></li><li class=nav-item><a class=nav-link href=/#about><span>Team</span></a></li><li class=nav-item><a class=nav-link href=/#publications><span>Papers</span></a></li><li class=nav-item><a class=nav-link href=/#projects><span>Projects</span></a></li><li class=nav-item><a class=nav-link href=/#talks><span>Talks</span></a></li><li class=nav-item><a class=nav-link href=/#posts><span>Posts</span></a></li><li class=nav-item><a class=nav-link href=/#gallery><span>Photos</span></a></li><li class=nav-item><a class=nav-link href=/#contact><span>Contact</span></a></li></ul></div><ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2"><li class=nav-item><a class="nav-link js-search" href=#><i class="fas fa-search" aria-hidden=true></i></a></li><li class=nav-item><a class="nav-link js-dark-toggle" href=#><i class="fas fa-moon" aria-hidden=true></i></a></li></ul></div></nav><div class=pub><div class="article-container pt-3"><h1>From sequence to signature: Uncovering multiscale AMR features across bacterial pathogens with supervised machine learning</h1><div class=article-metadata><div><span><a href=/authors/abhirupa-ghosh/>Abhirupa Ghosh</a></span>, <span><a href=/authors/evan-brenner/>Evan Brenner</a></span>, <span><a href=/authors/charmie-vang/>Charmie Vang</a></span>, <span><a href=/authors/ethan-wolfe/>Ethan Wolfe</a></span>, <span><a href=/authors/joseph-burke/>Joseph Burke</a></span>, <span><a href=/authors/raymond-lesiyon/>Raymond L Lesiyon</a></span>, <span><a href=/authors/keenan-manpearl/>Keenan Manpearl</a></span>, <span><a href=/authors/vignesh-sridhar/>Vignesh Sridhar</a></span>, <span><a href=/authors/jacob-krol/>Jacob Krol</a></span>, <span><a href=/authors/emily-boyer/>Emily Boyer</a></span>, <span><a href=/authors/jill-bilodeaux/>Jill Bilodeaux</a></span>, <span><a href=/authors/karn-jongnarangsin/>Karn Jongnarangsin</a></span>, <span><a href=/authors/elliot-majlessi/>Elliot Majlessi</a></span>, <span><a href=/authors/admin/>Janani Ravi</a></span></div><span class=article-date>July 2025</span>
<span class=middot-divider></span>
<span class=article-categories><i class="fas fa-folder mr-1"></i><a href=/categories/research/>Research</a></span></div><div class="btn-links mb-3"><a class="btn btn-outline-primary my-1 mr-1" href=https://doi.org/10.1101/2025.07.03.663053 target=_blank rel=noopener>Preprint</a>
<a class="btn btn-outline-primary my-1 mr-1" href=https://www.biorxiv.org/content/10.1101/2025.07.03.663053v1.full.pdf target=_blank rel=noopener>PDF</a>
<a class="btn btn-outline-primary my-1 mr-1" href=https://doi.org/10.1101/2025.07.03.663053 target=_blank rel=noopener>DOI</a>
<a class="btn btn-outline-primary my-1 mr-1" href=http://jravilab.org/amr target=_blank rel=noopener>Dashboard</a></div></div><div class="article-header article-container featured-image-wrapper mt-4 mb-4" style=max-width:720px;max-height:434px><div style=position:relative><img src=/publication/amr_eskape/featured_hu69322e731148510beb79bdb9d10ae1dc_1057860_720x0_resize_lanczos_3.png alt class=featured-image></div></div><div class=article-container><h3>Abstract</h3><p class=pub-abstract>Since the clinical introduction of antibiotics in the 1940s, antimicrobial resistance (AMR) has become an increasingly dire threat to global public health. Pathogens acquire AMR much faster than we discover new drugs, warranting new methods to better understand the molecular underpinnings of AMR. Traditional approaches for detecting AMR in novel bacterial strains are time-consuming and labor-intensive. However, advances in sequencing technology offer a plethora of bacterial genome data, and computational approaches like machine learning (ML) provide an optimistic scope for in silico AMR prediction. Here, we introduce a comprehensive multiscale ML approach to predict AMR phenotypes and identify AMR molecular features associated with a single drug or drug family, stratified by time and geographical locations. As a case study, we focus on a subset of the World Health Organization’s Bacterial Priority Pathogens, frequently drug-resistant and nosocomial bacteria: Enterococcus faecium, Staphylococcus aureus, Acinetobacter baumannii, Pseudomonas aeruginosa, and two Enterobacter species. We started with sequenced genomes with lab-derived AMR phenotypes, constructed pangenomes, clustered protein sequences, and extracted protein domains to generate pangenomic features across molecular scales. To uncover the molecular mechanisms behind drug-/drug family-specific resistance, we trained logistic regression ML models on our datasets. These yielded ranked lists of AMR-associated genes, proteins, and domains. In addition to recapitulating known AMR features, our models identified novel candidates for experimental validation. They performed well across molecular scales, data types (binary vs. counts), and drugs. The models achieved a median normalized Matthews correlation coefficient of 0.89, suggesting robust performance across molecular scales with counts or presence/absence of features. Model performance showed resilience even when evaluated on geographical and temporal holdouts. Our holistic approach promises reliable prediction of existing and developing resistance in newly sequenced pathogen genomes, along with the mechanistic molecular contributors of AMR.</p><div class=row><div class=col-md-1></div><div class=col-md-10><div class=row><div class="col-12 col-md-3 pub-row-heading">Type</div><div class="col-12 col-md-9"><a href=/publication/#3>Preprint</a></div></div></div><div class=col-md-1></div></div><div class="d-md-none space-below"></div><div class=row><div class=col-md-1></div><div class=col-md-10><div class=row><div class="col-12 col-md-3 pub-row-heading">Publication</div><div class="col-12 col-md-9"><em>bioRxiv</em></div></div></div><div class=col-md-1></div></div><div class="d-md-none space-below"></div><div class=space-below></div><div class=article-style><p>AG, EB, CV, EW are co-primary authors who contributed equally.</p></div><div class=article-tags><a class="badge badge-light" href=/tags/antimicrobial-resistance/>antimicrobial-resistance</a>
<a class="badge badge-light" href=/tags/pathogenic-features/>pathogenic-features</a>
<a class="badge badge-light" href=/tags/molecular-evolution/>molecular-evolution</a>
<a class="badge badge-light" href=/tags/machine-learning/>machine-learning</a>
<a class="badge badge-light" href=/tags/webapp/>webapp</a>
<a class="badge badge-light" href=/tags/compbio-bioinfo/>compbio-bioinfo</a></div><div class="media author-card content-widget-hr"><img class="avatar mr-3 avatar-circle" src=/authors/abhirupa-ghosh/avatar_huf93e19926ec90b6d2dc6153a0d3b7180_745776_270x270_fill_q90_lanczos_center.jpg alt=Avatar><div class=media-body><h5 class=card-title><a href=/authors/abhirupa-ghosh/>Abhirupa Ghosh</a></h5><h6 class=card-subtitle>Postdoctoral Researcher</h6><ul class=network-icon aria-hidden=true><li><a href=https://github.com/AbhirupaGhosh target=_blank rel=noopener><i class="fab fa-github"></i></a></li><li><a href=https://www.linkedin.com/in/abhirupa-ghosh-19169277/ target=_blank rel=noopener><i class="fab fa-linkedin"></i></a></li><li><a href=/files/ag-cv.pdf><i class="ai ai-cv"></i></a></li></ul></div></div><div class="article-widget content-widget-hr"><h3>Related</h3><ul><li><a href=/publication/molevolvr_webapp/>MolEvolvR: a web-app for characterizing proteins using molecular evolution and phylogeny</a></li><li><a href=/publication/psp_evolution/>The phage shock protein (PSP) envelope stress response: discovery of novel partners and evolutionary history</a></li><li><a href=/publication/amr_tim/>Unlocking antimicrobial resistance with multiomics and machine learning</a></li><li><a href=/publication/ggt_saureus/>The glutathione import system satisfies the Staphylococcus aureus nutrient sulfur requirement and promotes interspecies competition</a></li><li><a href=/talk/2022-ismb/>Cross-database integration using evolution and machine learning to identify multiscale molecular building blocks for antibiotic resistance</a></li></ul></div></div></div><script src=https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js integrity="sha256-eOgo0OtLL4cdq7RdwRUiGKLX9XsIJ7nGhWEKbohmVAQ=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/r.min.js></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.js integrity="sha256-EErZamuLefUnbMBQbsEqu1USa+btR2oIlCpBJbyD4/g=" crossorigin=anonymous></script>
<script>const code_highlighting=!0</script><script>const isSiteThemeDark=!0</script><script>const search_config={indexURI:"/index.json",minLength:1,threshold:.3},i18n={no_results:"No results found",placeholder:"Search...",results:"results found"},content_type={post:"Posts",project:"Projects",publication:"Publications",talk:"Talks"}</script><script id=search-hit-fuse-template type=text/x-template>
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script><script src=https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin=anonymous></script>
<script src=https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin=anonymous></script>
<script src=/js/academic.min.df22e735169cca39ea07a999d8cba15f.js></script><div class=container><footer class=site-footer><p class=powered-by>© Janani Ravi, 2025 &#183;
Powered by the
<a href=https://sourcethemes.com/academic/ target=_blank rel=noopener>Academic theme</a> for
<a href=https://gohugo.io target=_blank rel=noopener>Hugo</a>.
<span class=float-right aria-hidden=true><a href=# class=back-to-top><span class=button_icon><i class="fas fa-chevron-up fa-2x"></i></span></a></span></p></footer></div><div id=modal class="modal fade" role=dialog><div class=modal-dialog><div class=modal-content><div class=modal-header><h5 class=modal-title>Cite</h5><button type=button class=close data-dismiss=modal aria-label=Close>
<span aria-hidden=true>&#215;</span></button></div><div class=modal-body><pre><code class="tex hljs"></code></pre></div><div class=modal-footer><a class="btn btn-outline-primary my-1 js-copy-cite" href=# target=_blank><i class="fas fa-copy"></i> Copy</a>
<a class="btn btn-outline-primary my-1 js-download-cite" href=# target=_blank><i class="fas fa-download"></i> Download</a><div id=modal-error></div></div></div></div></div></body></html>