<!DOCTYPE html>
<html lang="en-us">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.7.0">

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Janani Ravi">

  
  
  
    
  
  <meta name="description" content="Background: Studying bacterial physiology, adaptation, and pathogenicity through the lens of evolution requires delineating the phylogenetic history of bacterial proteins and genomes. Moreover, delineating this history of proteins is best done at all three levels, sequence, structure, and function, and comparative pangenomics. However, currently, there are no unified frameworks that experimentalists can avail to seamlessly perform all these analyses for their proteins of their interest. Using these software/tools requires numerous skills/resources: install individual tools; download/preprocess massive datasets; match disparate input-output data formats across different tools; computational resources for efficient analyses; computational know-how to run command-line tools; intuitive result summarization and visualization.
Approach: To address this challenge, we have developed a computational framework for comprehensive evolutionary analysis that systematically integrates multiple data sources for gleaning sequence-structure-function relationships. Our framework goes beyond simple sequence comparisons by delving into constituent domains, domain architectures, phyletic spreads, and tracing the evolution across the tree of life. By adding a critical step of comparative pangenomics to these analyses, we can pinpoint molecular and genomic features that are unique to bacterial groups of interest (e.g., specific pathogens), which can then help prioritize candidate molecular targets, even in poorly characterized bacterial genomes.
Results: We have developed a computational approach for molecular evolution and phylogeny that researchers can perform homology searches across the tree of life and reconstructs domain architecture by characterizing the input proteins and each of their homologs by combining: sequence alignment and clustering algorithms for domain detection; profile matching against protein domain/orthology databases; and prediction algorithms for signal peptides, transmembrane regions, cellular localization, and secondary/tertiary structures. We are adding an additional layer of comparative pangenomics to identify unique features for bacterial groups of interest (e.g., pathogens vs nonpathogens). An instance of the web-app, applied to study a large number of Psp stress response proteins (present across the tree of life) can be found here: jravilab.shinyapps.io/psp-evolution. The web-app described here is a general web-server that researchers can use to study any protein(s) of interest. To demonstrate the versatility of this framework, we are currently applying it to zoonotic pathogens causing severe and chronic pathologies in humans and animals, for e.g., host-specific factors in Nontuberculous Mycobacteria, nutrient acquisition systems in Staphylococcus aureus, and surface layer proteins in Bacillus anthracis. We have implemented this computational approach for molecular evolution and phylogeny (MolEvolvR) as open-source software (R-package) and a streamlined easy-to-use web application (jraviilab.org/molevolvr) that will enable us and other researchers to prioritize candidate genetic factors in their application of interest for experimental validation.
">

  
  <link rel="alternate" hreflang="en-us" href="/talk/2021-glbio/">

  


  
  
  
  <meta name="theme-color" content="hsl(339, 90%, 68%)">
  

  
  
  
  <script src="/js/mathjax-config.js"></script>
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.0-1/css/all.min.css" integrity="sha256-4w9DunooKSr3MFXHXWyFER38WmPdm361bQS/2KUWZbU=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css" integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/github.min.css" crossorigin="anonymous" title="hl-light" disabled>
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/dracula.min.css" crossorigin="anonymous" title="hl-dark">
        
      
    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.css" integrity="sha256-SHMGCYmST46SoyGgo4YR/9AlK1vf3ff84Aq9yK4hdqM=" crossorigin="anonymous">
    

    

    
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.1.2/lazysizes.min.js" integrity="sha256-Md1qLToewPeKjfAHU1zyPwOutccPAm5tahnaw7Osw0A=" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js" integrity="" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    

  

  
  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap">
  

  
  
  
  
  <link rel="stylesheet" href="/css/academic.css">

  




  


  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_32x32_fill_lanczos_center_3.png">
  <link rel="apple-touch-icon" type="image/png" href="/images/icon_hu0b7a4cb9992c9ac0e91bd28ffd38dd00_9727_192x192_fill_lanczos_center_3.png">

  <link rel="canonical" href="/talk/2021-glbio/">

  
  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@JRaviLab">
  <meta property="twitter:creator" content="@JRaviLab">
  
  <meta property="og:site_name" content="JRavi Lab">
  <meta property="og:url" content="/talk/2021-glbio/">
  <meta property="og:title" content="Using a computational molecular evolution and phylogeny to study pathogenic proteins | JRavi Lab">
  <meta property="og:description" content="Background: Studying bacterial physiology, adaptation, and pathogenicity through the lens of evolution requires delineating the phylogenetic history of bacterial proteins and genomes. Moreover, delineating this history of proteins is best done at all three levels, sequence, structure, and function, and comparative pangenomics. However, currently, there are no unified frameworks that experimentalists can avail to seamlessly perform all these analyses for their proteins of their interest. Using these software/tools requires numerous skills/resources: install individual tools; download/preprocess massive datasets; match disparate input-output data formats across different tools; computational resources for efficient analyses; computational know-how to run command-line tools; intuitive result summarization and visualization.
Approach: To address this challenge, we have developed a computational framework for comprehensive evolutionary analysis that systematically integrates multiple data sources for gleaning sequence-structure-function relationships. Our framework goes beyond simple sequence comparisons by delving into constituent domains, domain architectures, phyletic spreads, and tracing the evolution across the tree of life. By adding a critical step of comparative pangenomics to these analyses, we can pinpoint molecular and genomic features that are unique to bacterial groups of interest (e.g., specific pathogens), which can then help prioritize candidate molecular targets, even in poorly characterized bacterial genomes.
Results: We have developed a computational approach for molecular evolution and phylogeny that researchers can perform homology searches across the tree of life and reconstructs domain architecture by characterizing the input proteins and each of their homologs by combining: sequence alignment and clustering algorithms for domain detection; profile matching against protein domain/orthology databases; and prediction algorithms for signal peptides, transmembrane regions, cellular localization, and secondary/tertiary structures. We are adding an additional layer of comparative pangenomics to identify unique features for bacterial groups of interest (e.g., pathogens vs nonpathogens). An instance of the web-app, applied to study a large number of Psp stress response proteins (present across the tree of life) can be found here: jravilab.shinyapps.io/psp-evolution. The web-app described here is a general web-server that researchers can use to study any protein(s) of interest. To demonstrate the versatility of this framework, we are currently applying it to zoonotic pathogens causing severe and chronic pathologies in humans and animals, for e.g., host-specific factors in Nontuberculous Mycobacteria, nutrient acquisition systems in Staphylococcus aureus, and surface layer proteins in Bacillus anthracis. We have implemented this computational approach for molecular evolution and phylogeny (MolEvolvR) as open-source software (R-package) and a streamlined easy-to-use web application (jraviilab.org/molevolvr) that will enable us and other researchers to prioritize candidate genetic factors in their application of interest for experimental validation.
"><meta property="og:image" content="/talk/2021-glbio/featured.png">
  <meta property="twitter:image" content="/talk/2021-glbio/featured.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2021-05-13T00:00:00&#43;00:00">
    
    <meta property="article:modified_time" content="2021-05-13T20:00:00&#43;00:00">
  

  



  
    




<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "Event",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "/talk/2021-glbio/"
  },
  "name": "Using a computational molecular evolution and phylogeny to study pathogenic proteins",
  
  "location": {
    "@type": "Place",
    "name": "Virtual"
  },
  
  
  "image": [
    "/talk/2021-glbio/featured.png"
  ],
  
  "startDate": "2021-05-13T20:00:00Z",
  
  "endDate": "2021-05-13T21:00:00Z",
  
  
  "performer": {
    "@type": "Person",
    "name": "Janani Ravi"
  },
  
  "description": "Background: Studying bacterial physiology, adaptation, and pathogenicity through the lens of evolution requires delineating the phylogenetic history of bacterial proteins and genomes. Moreover, delineating this history of proteins is best done at all three levels, sequence, structure, and function, and comparative pangenomics. However, currently, there are no unified frameworks that experimentalists can avail to seamlessly perform all these analyses for their proteins of their interest. Using these software/tools requires numerous skills/resources: install individual tools; download/preprocess massive datasets; match disparate input-output data formats across different tools; computational resources for efficient analyses; computational know-how to run command-line tools; intuitive result summarization and visualization.\nApproach: To address this challenge, we have developed a computational framework for comprehensive evolutionary analysis that systematically integrates multiple data sources for gleaning sequence-structure-function relationships. Our framework goes beyond simple sequence comparisons by delving into constituent domains, domain architectures, phyletic spreads, and tracing the evolution across the tree of life. By adding a critical step of comparative pangenomics to these analyses, we can pinpoint molecular and genomic features that are unique to bacterial groups of interest (e.g., specific pathogens), which can then help prioritize candidate molecular targets, even in poorly characterized bacterial genomes.\nResults: We have developed a computational approach for molecular evolution and phylogeny that researchers can perform homology searches across the tree of life and reconstructs domain architecture by characterizing the input proteins and each of their homologs by combining: sequence alignment and clustering algorithms for domain detection; profile matching against protein domain/orthology databases; and prediction algorithms for signal peptides, transmembrane regions, cellular localization, and secondary/tertiary structures. We are adding an additional layer of comparative pangenomics to identify unique features for bacterial groups of interest (e.g., pathogens vs nonpathogens). An instance of the web-app, applied to study a large number of Psp stress response proteins (present across the tree of life) can be found here: jravilab.shinyapps.io/psp-evolution. The web-app described here is a general web-server that researchers can use to study any protein(s) of interest. To demonstrate the versatility of this framework, we are currently applying it to zoonotic pathogens causing severe and chronic pathologies in humans and animals, for e.g., host-specific factors in Nontuberculous Mycobacteria, nutrient acquisition systems in Staphylococcus aureus, and surface layer proteins in Bacillus anthracis. We have implemented this computational approach for molecular evolution and phylogeny (MolEvolvR) as open-source software (R-package) and a streamlined easy-to-use web application (jraviilab.org/molevolvr) that will enable us and other researchers to prioritize candidate genetic factors in their application of interest for experimental validation.\n"
}
</script>

  


  


  





  <title>Using a computational molecular evolution and phylogeny to study pathogenic proteins | JRavi Lab</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="dark">

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  







<nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
    <div class="d-none d-lg-inline-flex">
      <a class="navbar-brand" href="/">JRavi Lab</a>
    </div>
    

    
    <button type="button" class="navbar-toggler" data-toggle="collapse"
            data-target="#navbar-content" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
    <span><i class="fas fa-bars"></i></span>
    </button>
    

    
    <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
      <a class="navbar-brand" href="/">JRavi Lab</a>
    </div>
    

    
    
    <div class="navbar-collapse main-menu-item collapse justify-content-start" id="navbar-content">

      
      <ul class="navbar-nav d-md-inline-flex">
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about_group"><span>Home</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#interests"><span>Interests</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#news"><span>News</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>PI</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#people"><span>Team</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#publications"><span>Papers</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#talks"><span>Talks</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>Posts</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#gallery"><span>Photos</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#contact"><span>Contact</span></a>
        </li>

        
        

      

        
      </ul>
    </div>

    <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">
      
      <li class="nav-item">
        <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
      </li>
      

      
      <li class="nav-item">
        <a class="nav-link js-dark-toggle" href="#"><i class="fas fa-moon" aria-hidden="true"></i></a>
      </li>
      

      

    </ul>

  </div>
</nav>


  <div class="pub">

  




















  
  


<div class="article-container pt-3">
  <h1>Using a computational molecular evolution and phylogeny to study pathogenic proteins</h1>

  

  


<div class="article-metadata">

  
  
  
  
  <div>
    



  
  <span><a href="/authors/janani-ravi/">Janani Ravi</a></span>

  </div>
  
  

  

  

  

  
  
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder mr-1"></i><a href="/categories/research/">Research</a>, <a href="/categories/r/">R</a></span>
  

</div>

  













<div class="btn-links mb-3">
  
  








  









  
  <a class="btn btn-outline-primary my-1 mr-1" href="/project/pathogenic-features/">
    Project
  </a>
  

  
  <a class="btn btn-outline-primary my-1 mr-1" href="/project/stress-response/">
    Project
  </a>
  





  
  
    
  
<a class="btn btn-outline-primary my-1 mr-1" href="https://docs.google.com/presentation/d/e/2PACX-1vQ4mq7cRkBshvp27yBSOZcRxyeTrG-Oe7Fs8f8BoMNYStr5zTweHU4it9GylyORkE45KVPVTpgcFAof/pub?start=false&amp;loop=false&amp;delayms=3000" target="_blank" rel="noopener">
  Slides
</a>







</div>


</div>


<div class="article-header article-container featured-image-wrapper mt-4 mb-4" style="max-width: 231px; max-height: 234px;">
  <div style="position: relative">
    <img src="/talk/2021-glbio/featured.png" alt="" class="featured-image">
    
  </div>
</div>



  <div class="article-container">

    
      <h3>Abstract</h3>
      <p class="pub-abstract">Background: Studying bacterial physiology, adaptation, and pathogenicity through the lens of evolution requires delineating the phylogenetic history of bacterial proteins and genomes. Moreover, delineating this history of proteins is best done at all three levels, sequence, structure, and function, and comparative pangenomics. However, currently, there are no unified frameworks that experimentalists can avail to seamlessly perform all these analyses for their proteins of their interest. Using these software/tools requires numerous skills/resources: install individual tools; download/preprocess massive datasets; match disparate input-output data formats across different tools; computational resources for efficient analyses; computational know-how to run command-line tools; intuitive result summarization and visualization.
Approach: To address this challenge, we have developed a computational framework for comprehensive evolutionary analysis that systematically integrates multiple data sources for gleaning sequence-structure-function relationships. Our framework goes beyond simple sequence comparisons by delving into constituent domains, domain architectures, phyletic spreads, and tracing the evolution across the tree of life. By adding a critical step of comparative pangenomics to these analyses, we can pinpoint molecular and genomic features that are unique to bacterial groups of interest (e.g., specific pathogens), which can then help prioritize candidate molecular targets, even in poorly characterized bacterial genomes.
Results: We have developed a computational approach for molecular evolution and phylogeny that researchers can perform homology searches across the tree of life and reconstructs domain architecture by characterizing the input proteins and each of their homologs by combining: sequence alignment and clustering algorithms for domain detection; profile matching against protein domain/orthology databases; and prediction algorithms for signal peptides, transmembrane regions, cellular localization, and secondary/tertiary structures. We are adding an additional layer of comparative pangenomics to identify unique features for bacterial groups of interest (e.g., pathogens vs nonpathogens). An instance of the web-app, applied to study a large number of Psp stress response proteins (present across the tree of life) can be found here: jravilab.shinyapps.io/psp-evolution. The web-app described here is a general web-server that researchers can use to study any protein(s) of interest. To demonstrate the versatility of this framework, we are currently applying it to zoonotic pathogens causing severe and chronic pathologies in humans and animals, for e.g., host-specific factors in Nontuberculous Mycobacteria, nutrient acquisition systems in Staphylococcus aureus, and surface layer proteins in Bacillus anthracis. We have implemented this computational approach for molecular evolution and phylogeny (MolEvolvR) as open-source software (R-package) and a streamlined easy-to-use web application (jraviilab.org/molevolvr) that will enable us and other researchers to prioritize candidate genetic factors in their application of interest for experimental validation.</p>
    

    <div class="row">
      <div class="col-md-1"></div>
      <div class="col-md-10">
        <div class="row">
          <div class="col-12 col-md-3 pub-row-heading">Date</div>
          <div class="col-12 col-md-9">
            May 13, 2021 8:00 PM &mdash; 9:00 PM
          </div>
        </div>
      </div>
      <div class="col-md-1"></div>
    </div>
    <div class="d-md-none space-below"></div>

    
    <div class="row">
      <div class="col-md-1"></div>
      <div class="col-md-10">
        <div class="row">
          <div class="col-12 col-md-3 pub-row-heading">Event</div>
          <div class="col-12 col-md-9">
            <a href="http://iscb.org/cms_addon/conferences/glbio2021/tracks/General" target="_blank" rel="noopener">
            Great Lakes Bioinformatics Conference (GLBIO) 2021
            </a>
          </div>
        </div>
      </div>
      <div class="col-md-1"></div>
    </div>
    <div class="d-md-none space-below"></div>
    

    
    <div class="row">
      <div class="col-md-1"></div>
      <div class="col-md-10">
        <div class="row">
          <div class="col-12 col-md-3 pub-row-heading">Location</div>
          <div class="col-12 col-md-9">Virtual</div>
          
        </div>
      </div>
      <div class="col-md-1"></div>
    </div>
    <div class="d-md-none space-below"></div>
    

    <div class="space-below"></div>

    <div class="article-style">
      <p>Related: Samuel Z Chen GLBio poster [video])(
<a href="https://www.youtube.com/watch?v=6rj3eTWxXkg" target="_blank" rel="noopener">https://www.youtube.com/watch?v=6rj3eTWxXkg</a>)</p>

    </div>

    





<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/compbio-bioinfo/">compbio-bioinfo</a>
  
  <a class="badge badge-light" href="/tags/molecular-evolution/">molecular-evolution</a>
  
  <a class="badge badge-light" href="/tags/pathogenic-features/">pathogenic-features</a>
  
  <a class="badge badge-light" href="/tags/phylogeny/">phylogeny</a>
  
  <a class="badge badge-light" href="/tags/stress-response/">stress-response</a>
  
  <a class="badge badge-light" href="/tags/comparative-genomics/">comparative-genomics</a>
  
</div>














  
  
    
  
  






  
  
  
  
  
  <div class="media author-card content-widget-hr">
    

    <div class="media-body">
      <h5 class="card-title"><a href="/authors/janani-ravi/"></a></h5>
      
      
      <ul class="network-icon" aria-hidden="true">
  
</ul>

    </div>
  </div>









  
  
  <div class="article-widget content-widget-hr">
    <h3>Related</h3>
    <ul>
      
      <li><a href="/talk/2021-beacon/">MolEvolvR: a web-app for molecular evolution and phylogeny</a></li>
      
      <li><a href="/project/pathogenic-features/">Computational evolutionary approaches</a></li>
      
      <li><a href="/publication/psp_evolution/">Phage-shock-protein (Psp) Envelope Stress Response: Evolutionary History &amp; Discovery of Novel Players.</a></li>
      
      <li><a href="/talk/2020-beacon/">Computational evolutionary approaches for understanding pathogen proteins and genomes</a></li>
      
      <li><a href="/talk/2020-ismb/">Computational evolutionary approaches to characterize bacterial proteins</a></li>
      
    </ul>
  </div>
  



  </div>
</div>

      

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js" integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js" integrity="sha256-eOgo0OtLL4cdq7RdwRUiGKLX9XsIJ7nGhWEKbohmVAQ=" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/languages/r.min.js"></script>
        
      

    

    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.5.1/leaflet.js" integrity="sha256-EErZamuLefUnbMBQbsEqu1USa+btR2oIlCpBJbyD4/g=" crossorigin="anonymous"></script>
    

    
    
    <script>const code_highlighting = true;</script>
    

    
    
    <script>const isSiteThemeDark = true;</script>
    

    
    
    
    
    
    
    <script>
      const search_config = {"indexURI":"/index.json","minLength":1,"threshold":0.3};
      const i18n = {"no_results":"No results found","placeholder":"Search...","results":"results found"};
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.05ad38647aed3ac5a5800d9026541eee.js"></script>

    






  
  
  <div class="container">
    <footer class="site-footer">
  

  <p class="powered-by">
    Â© Janani Ravi, 2022 &middot; 

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" class="back-to-top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
